lib/absinthe/phase/document/execution/resolution.ex

```diff
@@ -143,6 +143,17 @@ defmodule Absinthe.Phase.Document.Execution.Resolution do
     |> to_result(bp_field, full_type)
     |> walk_result(acc, bp_field, full_type, info)
   end
+  defp build_result({:error, %Error{} = err}, acc, bp_field, info, _) do
```

Comment:
I would rather it's always a 2-tuple. Rather than "extra" fields, if you want to have a more stuff, provide a Keyword.t or map (or possibly a list of them) that must have :message in it.